code/tests/btrfs_link_unlink.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+Reproducing new btrfs bug reported
+
+1. Create directory A under TEST_MNT
+2. Create file foo in dir A
+3. Create a link bar to file foo in the same dir A
+4. Sync the changes
+5. Unlink bar
+6. Create file bar again
+7. fsync(bar)
+
+If a power fail occurs now, the file system must be mountable
+
+
+Fails on btrfs kernel 4.16. The filesystem becomes unmountable
+
+https://marc.info/?l=linux-btrfs&m=151983337811947&w=2
+https://marc.info/?l=linux-btrfs&m=151983349112005&w=2
+*/
+
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string>
+#include <iostream>
+#include <dirent.h>
+#include <cstring>
+#include <errno.h>
+
+#include "BaseTestCase.h"
+#include "../user_tools/api/workload.h"
+#include "../user_tools/api/actions.h"
+#define TEST_FILE_FOO "foo"
+#define TEST_FILE_BAR "bar"
+#define TEST_MNT "/mnt/snapshot"
```

Comment:
Can you use the variable `mnt_dir_` instead of this define? You'll have to rewrite where you concatenate things since `mnt_dir_` is string. https://github.com/utsaslab/crashmonkey/blob/master/code/tests/BaseTestCase.h#L21